/*
 * This file was automatically generated by EvoSuite
 * Mon Oct 28 22:45:36 GMT 2019
 */

package module;

import org.junit.Test;
import static org.junit.Assert.*;
import static org.evosuite.runtime.EvoAssertions.*;
import java.sql.SQLIntegrityConstraintViolationException;
import java.sql.SQLTimeoutException;
import module.Message;
import module.MessageProcessingException;
import module.Module;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.junit.runner.RunWith;

@RunWith(EvoRunner.class) @EvoRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true, separateClassLoader = true, useJEE = true) 
public class MessageProcessingException_ESTest extends MessageProcessingException_ESTest_scaffolding {

  @Test(timeout = 4000)
  public void test0()  throws Throwable  {
      Message message0 = new Message();
      SQLTimeoutException sQLTimeoutException0 = new SQLTimeoutException();
      MessageProcessingException messageProcessingException0 = new MessageProcessingException((Module) null, message0, "util.TaintedByteArrayInputStream", sQLTimeoutException0);
  }

  @Test(timeout = 4000)
  public void test1()  throws Throwable  {
      Message message0 = new Message();
      message0.setBody(message0);
      MessageProcessingException messageProcessingException0 = null;
      try {
        messageProcessingException0 = new MessageProcessingException(message0, "");
        fail("Expecting exception: StackOverflowError");
      
      } catch(StackOverflowError e) {
         //
         // no message in exception (getMessage() returned null)
         //
      }
  }

  @Test(timeout = 4000)
  public void test2()  throws Throwable  {
      Message message0 = new Message();
      message0.setBody(message0);
      MessageProcessingException messageProcessingException0 = null;
      try {
        messageProcessingException0 = new MessageProcessingException(message0);
        fail("Expecting exception: StackOverflowError");
      
      } catch(StackOverflowError e) {
         //
         // no message in exception (getMessage() returned null)
         //
      }
  }

  @Test(timeout = 4000)
  public void test3()  throws Throwable  {
      MessageProcessingException messageProcessingException0 = new MessageProcessingException("", (Throwable) null);
  }

  @Test(timeout = 4000)
  public void test4()  throws Throwable  {
      MessageProcessingException messageProcessingException0 = new MessageProcessingException((Throwable) null);
  }

  @Test(timeout = 4000)
  public void test5()  throws Throwable  {
      Message message0 = new Message();
      MessageProcessingException messageProcessingException0 = new MessageProcessingException(message0);
  }

  @Test(timeout = 4000)
  public void test6()  throws Throwable  {
      Message message0 = new Message();
      MessageProcessingException messageProcessingException0 = new MessageProcessingException(message0, "msg=");
  }

  @Test(timeout = 4000)
  public void test7()  throws Throwable  {
      Message message0 = new Message();
      SQLIntegrityConstraintViolationException sQLIntegrityConstraintViolationException0 = new SQLIntegrityConstraintViolationException();
      MessageProcessingException messageProcessingException0 = new MessageProcessingException((Module) null, message0, sQLIntegrityConstraintViolationException0);
  }

  @Test(timeout = 4000)
  public void test8()  throws Throwable  {
      Message message0 = new Message();
      SQLIntegrityConstraintViolationException sQLIntegrityConstraintViolationException0 = new SQLIntegrityConstraintViolationException();
      MessageProcessingException messageProcessingException0 = new MessageProcessingException(message0, sQLIntegrityConstraintViolationException0);
  }
}
